src/component-library/utils/prop-types.ts

```diff
@@ -1,7 +1,11 @@
-const tuple = <T extends string[]>(...args: T): string[] => args;
+const tuple = <T extends string[]>(...args: T): T => args;
 
 const ctaVariants = tuple('primary', 'secondary', 'outlined');
 
+const severity = tuple('error', 'warning', 'success');
+
 export type CTAVariants = typeof ctaVariants[number];
 
+export type Severity = typeof severity[number];
```

Comment:
Its good to have this types declared as ts variables. You could later use them to do checks suchs as `severity.includes(X)`, instead of every time creating the array. This also helps us to document our storybook later on.